clippy_lints/src/literal_representation.rs

```diff
@@ -414,20 +414,29 @@ impl LiteralDigitGrouping {
         radix: Radix,
         lint_unreadable: bool,
     ) -> Result<Option<usize>, WarningType> {
-        let mut groups = groups.map(str::len);
+        let groups = groups.collect::<Vec<_>>();
+        assert!(!groups.is_empty());
```

Comment:
I don't really like `assert!`s in the compiler unless we are 200% sure that it cannot trigger. Could you change it or make a comment rationalizing the decision?

Reply:
I did like the original code (which was using `expect`) but actually, even that was not great, so I'll you `let else` instead.